db/data/20221003201116_populate_default_role_settings.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class PopulateDefaultRoleSettings < ActiveRecord::Migration[7.0]
+  def up
+    Setting.create! [
+      { name: 'DefaultRole' }
+    ]
+
+    SiteSetting.create! [
+      { setting: Setting.find_by(name: 'DefaultRole'), provider: 'greenlight', value: 'User' }
+    ]
+  end
+
+  def down
+    Setting.find_by(name: 'DefaultRole').delete
```

Comment:
I suspect that this will fail,
A general rule of thumb, destroy the dependent resources before destroying the resource that uses them.

Reply:
What do you propose? Delete all Settings and all Roles?
Most of our recent migrations have `ActiveRecord::IrreversibleMigration`

Reply:
I swapped them

Reply:
I think the issue isn't solved,
You can try rolling back the data migration to confirm if your rollback logic is valid for that you can run `rails data:rollback` simply since it's the last migration,
I believe something like was intended:
```
  def down
    setting = Setting.find_by(name: 'DefaultRole')
    SiteSetting.where(setting:).destroy_all
    setting.destroy!
  end
```

Reply:
Got it. Nicely done. Thank you